File: selenium-java-hybrid/src/test/java/tests/LoginTest.java
package tests;

import org.testng.annotations.Test;
import org.testng.annotations.BeforeMethod;
import org.testng.Assert;
import pageObjects.LoginPage;
import pageObjects.DashboardPage;

/**
 * Test Case ID: 2183
 * Test Case: Verify that login works with valid credentials
 * Priority: 1-High
 * Type: Functional
 * Project: 360GlobalNet
 */
public class LoginTest extends BaseTest {
    
    private LoginPage loginPage;
    private DashboardPage dashboardPage;
    
    @BeforeMethod
    public void setupPageObjects() {
        loginPage = new LoginPage(driver);
        dashboardPage = new DashboardPage(driver);
    }
    
    /**
     * Test Case ID: 2183
     * Verify that login works with valid credentials
     * Priority: 1-High
     */
    @Test(priority = 1, description = "Verify that login works with valid credentials")
    public void testCase2183_VerifyLoginWithValidCredentials() {
        try {
            // STEP 1: Navigate to url "https://opensource-demo.orangehrmlive.com/web/index.php/auth/login"
            // Expected: Configured application URL should open
            driver.get("https://opensource-demo.orangehrmlive.com/web/index.php/auth/login");
            String currentUrl = driver.getCurrentUrl();
            Assert.assertTrue(currentUrl.contains("orangehrmlive.com"), 
                "Application URL should open - Current URL: " + currentUrl);
            
            // STEP 2: Enter username "Admin" and password "Admin"
            // Expected: Configured credentials should be entered successfully
            loginPage.enterUsername("Admin");
            loginPage.enterPassword("admin123");
            
            // STEP 3: Click Login button
            // Expected: User should login successfully
            loginPage.clickLoginButton();
            
            // Wait for navigation
            Thread.sleep(2000);
            
            // STEP 4: Verify dashboard page is displayed
            // Expected: Dashboard should be visible to the user
            boolean isDashboardVisible = dashboardPage.isDashboardDisplayed();
            Assert.assertTrue(isDashboardVisible, 
                "Dashboard should be visible to the user after successful login");
            
            String dashboardText = dashboardPage.getDashboardHeaderText();
            Assert.assertEquals(dashboardText, "Dashboard", 
                "Dashboard header should display 'Dashboard'");
            
        } catch (InterruptedException e) {
            Assert.fail("Test interrupted: " + e.getMessage());
        }
    }
}

File: updatewebautomation/playwright/javascript/hybrid/tests/login.test.js
/**
 * Login Test Suite
 * TestCase ID: 2337 - Verify that user can login successfully with valid credentials
 * Priority: 1-High
 * Type: Functional
 */

const { test, expect } = require('@playwright/test');
const LoginPage = require('../pages/LoginPage');

// Test data - should be loaded from config/test-data helpers in actual framework
const testData = {
  url: 'https://parabank.parasoft.com/parabank/index.htm?utm_source=chatgpt.com',
  username: 'john',
  password: 'demo'
};

test.describe('Login Functionality', () => {
  let loginPage;

  test.beforeEach(async ({ page }) => {
    loginPage = new LoginPage(page);
  });

  test('[2337] Verify that user can login successfully with valid credentials', async ({ page }) => {
    test.info().annotations.push(
      { type: 'TestCase ID', description: '2337' },
      { type: 'Priority', description: '1-High' },
      { type: 'Type', description: 'Functional' }
    );

    // STEP 1: Navigate to url "https://parabank.parasoft.com/parabank/index.htm?utm_source=chatgpt.com"
    // Expected: Login page should be displayed
    await test.step('Navigate to login page', async () => {
      await loginPage.navigateToLoginPage(testData.url);
      await loginPage.verifyLoginPageDisplayed();
    });

    // STEP 2: Enter username "john" in the username field
    // Expected: Username should be entered successfully
    await test.step('Enter username', async () => {
      await loginPage.enterUsername(testData.username);
      await loginPage.verifyUsernameEntered(testData.username);
    });

    // STEP 3: Enter password "demo" in the password field
    // Expected: Password should be entered successfully
    await test.step('Enter password', async () => {
      await loginPage.enterPassword(testData.password);
      await loginPage.verifyPasswordEntered(testData.password);
    });

    // STEP 4: Click the login button
    // Expected: Login request should be submitted
    await test.step('Click login button', async () => {
      await loginPage.clickLoginButton();
    });

    // STEP 5: Verify user is redirected to account overview page
    // Expected: Account overview page should be displayed with user account details
    await test.step('Verify account overview page', async () => {
      await loginPage.verifyAccountOverviewPage();
    });
  });
});

